File: ProblemSet/70.climbing-stairs/python/70.climbing-stairs 公式法.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0:
            return 0
        gloden_ratio = (1 + 5 ** 0.5) / 2
        return round(gloden_ratio ** (n + 1) / (5 ** 0.5))
        
# @lc code=end

File: ProblemSet/70.climbing-stairs/java/70.climbing-stairs 公式法.java
/*
 * @lc app=leetcode id=70 lang=java
 *
 * [70] Climbing Stairs
 */

// @lc code=start
class Solution {
    public int climbStairs(int n) {
        double glodenRatio = (1 + Math.sqrt(5)) / 2;
        return (int)Math.round(Math.pow(glodenRatio, n + 1) / Math.sqrt(5));
    }
}
// @lc code=end

